### PR TITLE
newick writing support

### PR DESCRIPTION
**Pull request by wasade**
@kwcantrell, sorry to bother -- able to once over this one as well?

Note I just added GH actions and getting that working in master. Once thats good I'll merge the commits here

### CODE REVIEW

**Comment by wasade** [comment]
Ugh, bit array isn't coming through on actions. Unfortunately I'm out of time today to poke. WIll fix as soon as I can. Code works locally at least... :)

**Comment by wasade** [comment]
@kwcantrell, okay I think I fixed this finally... able to once over the newick writing pieces?

File: bp/_io.pyx [hunk 1]
```diff
@@ -36,6 +36,52 @@ cdef void _set_node_metadata(np.uint32_t ptr, unicode token,
...
+            name_stack.append(tree.name(idx))
+            length_stack.append(tree.length(idx))
+            open_paren_stack.append(idx)
+        else:
+            name = name_stack.pop()
+            length = length_stack.pop()
+
+            if name is not None:
+                # if we have magical characters, make sure we quote
+                if set(name) & {';', ',', '(', ')'}:
```
Comment (kwcantrell) [inline bp/_io.pyx:68]: Can ':' be a magical character?

Comment (wasade) [reply bp/_io.pyx:68]: oh snarky, ya good point

**Comment by kwcantrell** [comment]
Looks good. There might be other characters that we could check but I think those are the ones with special meaning behind them. If this becomes an issue we can probably do something similar along the lines of .isalnum()

File: bp/_io.pyx (hunk 1 above)
Comment (kwcantrell) [reply bp/_io.pyx]: Looks good. There might be other characters that we could check but I think those are the ones with special meaning behind them. If this becomes an issue we can probably do something similar along the lines of .isalnum()

Comment (wasade) [reply bp/_io.pyx]: +1

**Pull request merged**